core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -73,6 +73,7 @@
 /** Adaptor class to translate REST requests into {@link Catalog} API calls. */
 public class RESTCatalogAdapter implements RESTClient {
   private static final Splitter SLASH = Splitter.on('/');
+  private static final String NAMESPACE_SEPARATOR = "%2E";
```

Comment:
Why is this not `.`? It's not a character that needs to be escaped.

Reply:
`CatalogTests#testNamespaceWithDot` would fail when using `.` here, so in this case it needs to be the UTF-8 encoded string.